feat: job creation mode GA

shollyman opened the pull request:
This PR makes the underlying functionality related to how queries can optionally avoid job creation a GA feature.  It does the following:

* no longer uses the preview QUERY_PREVIEW_ENABLED environment variable to control job creation
* adds a new argument to Client instantiation to control job creation mode
* adds a property/setter to Client to control job creation mode

This PR also updates/renames the sample demonstrating how to leverage job creation mode with Client.query_and_wait.

internal: b/396766171

shollyman committed:
feat: job creation mode GA

This PR makes the underlying functionality related to how queries can
optionally avoid job creation a GA feature.  It does the following:

* no longer uses the preview QUERY_PREVIEW_ENABLED environment variable
  to control job creation
* adds a new argument to Client instantiation to control job creation
  mode
* adds a property/setter to Client to control job creation mode

This PR also updates/renames the sample demonstrating how to leverage
job creation mode with Client.query_and_wait.

shollyman committed:
linting

shollyman committed:
switch to prop setter

shollyman committed:
Merge branch 'main' into jco-ga

shollyman commented:
re: sample tag change, will do the freeze and rename dance on the docs side when we're ready.

tswast commented on google/cloud/bigquery/client.py:
```diff
@@ -286,6 +292,15 @@ def location(self):
+    @property
+    def default_job_creation_mode(self):
+        """Default job creation mode used for query execution."""
+        return self._default_job_creation_mode
```
Chatted offline. I'll aim to get bigframes 1.x and 2.x releases out to support this change before the client library release to minimize disruption.

Linchin commented on samples/client_query_job_optional.py:
```diff
@@ -44,10 +46,12 @@ def client_query_shortmode() -> None:
     else:
-        print("Query was run in short mode.  Query ID: {}".format(rows.query_id))
+        print(
+            "Query was run without creating a job..  Query ID: {}".format(rows.query_id)
```
```suggestion
            "Query was run without creating a job.  Query ID: {}".format(rows.query_id)
```

Linchin commented on tests/unit/test__job_helpers.py:
```diff
@@ -605,6 +601,7 @@ def test_query_and_wait_sets_job_creation_mode(monkeypatch: pytest.MonkeyPatch):
 
 def test_query_and_wait_sets_location():
     client = mock.create_autospec(Client)
+    client.default_job_creation_mode = None
```
Just for my education, why do we need to set `client.default_job_creation_mode = None` explicitly? It seems using default value would be enough.

shollyman replied:
Originally I tried to keep the asserted request unchanged, but I've instead made this less confusing and used mock.ANY for the jobCreationMode aspect of the request.

Linchin commented on google/cloud/bigquery/client.py:
```diff
@@ -286,6 +292,15 @@ def location(self):
+        """Default job creation mode used for query execution."""
+        return self._default_job_creation_mode
+
+    @default_job_creation_mode.setter
```
nit: add unit test for the property

shollyman replied:
done

chalmerlowe committed:
Merge branch 'main' into jco-ga

renovate-bot committed:
fix(deps): update all dependencies (#2184)

* fix(deps): update all dependencies

* Update pyproject.toml

* Update .github/workflows/docs.yml

* Update .github/workflows/docs.yml

---------

Co-authored-by: Chalmer Lowe <chalmerlowe@google.com>

chalmerlowe committed:
ddocs: fixes several typos and updates a try except block (#2197)

chalmerlowe committed:
ci: Remove unit tests and doc tests from kokoro presubmit. (#2195)

* The message is a commit message, not a message to a user from an AI agent. Therefore, it should be output as is.

Output:
Remove Kokoro presubmit for unit, docs, and coverage.

This commit removes the Kokoro presubmit configuration that runs
`unit_noextras`, `unit`, `cover`, `docs`, and `docfx` nox sessions.
These checks are already performed by GitHub Actions, making the
Kokoro configuration redundant.

The change involves removing the `NOX_SESSION` environment variable
definition from `.kokoro/presubmit/presubmit.cfg`.

* Update presubmit.cfg

* Delete .kokoro/presubmit/presubmit.cfg

---------

Co-authored-by: google-labs-jules[bot] <161369871+google-labs-jules[bot]@users.noreply.github.com>

chalmerlowe committed:
ci: Configure Renovate to keep Python at 3.10 for docs workflow (#2199)

* Configure Renovate to keep Python at 3.10 for docs workflow

This change adds a packageRule to `renovate.json` to prevent Renovate
from updating the Python version used in the `.github/workflows/docs.yml`
GitHub Actions workflow.

The rule specifically targets the `python-version` input of the
`actions/setup-python` step and restricts allowed versions to `<3.11`,
effectively pinning it to `3.10` for now.

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

* Update renovate.json

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

* adds files to excludes lists

* 🦉 Updates from OwlBot post-processor

See https://github.com/googleapis/repo-automation-bots/blob/main/packages/owl-bot/README.md

* Update owlbot.py

* adds packageRule about pyproject.toml

---------

Co-authored-by: google-labs-jules[bot] <161369871+google-labs-jules[bot]@users.noreply.github.com>
Co-authored-by: Owl Bot <gcf-owl-bot[bot]@users.noreply.github.com>

chalmerlowe committed:
ci: updates renovate.json to ignore docs.yml (#2200)

* updates renovate to ignore docs.yml

* Update renovate.json

renovate-bot committed:
chore(deps): update dependency google-auth to v2.40.2 (#2196)

tswast committed:
Update samples/client_query_job_optional.py

Co-authored-by: Lingqing Gan <lingqing.gan@gmail.com>

tswast committed:
docs: use query_and_wait in the array parameters sample (#2202)

shollyman committed:
address reviewer feedback

shollyman committed:
Merge branch 'main' into jco-ga

Linchin commented on samples/tests/test_client_query_job_optional.py:
```diff
@@ -14,13 +14,13 @@
 
 def test_client_query_shortmode(capsys: "pytest.CaptureFixture[str]") -> None:
-    client_query_shortmode.client_query_shortmode()
+    client_query_job_optional.client_query_job_optional()
```
Should we add some verification that the query was indeed run in jobless mode?

shollyman replied:
Control over execution is still the responsible of the service, and it can choose to fallback to using jobs regardless of the requested mode.
